snapcraft/internal/yaml.py

```diff
@@ -109,15 +98,20 @@ def __init__(self, part):
 
 class Config:
 
+    @property
+    def part_names(self):
+        return self.parts_config._part_names
```

Comment:
Great catch.  I've updated it to export from PartConfig and access it via that in Config.